### PR TITLE
[AIRAVATA-2652] Remove registry-core dependency on orchestrator and gfac modules 

### PR DESCRIPTION
**Pull request by sachinkariyattin**
Removed `airavata-registry-cpi` and `airavata-registry-core` dependencies from Gfac and Orchestrator modules and implemented changes to use `registry-api` module instead. Also added new methods to the registry api which were required by the gfac and orchestrator modules.

**Major Changes:**
- Added the following methods to the Registry API
    
        addExperimentProcessOutputs()
        addErrors()
        addTaskStatus()
        addProcessStatus()
        updateProcessStatus()
        updateExperimentStatus()
        addJobStatus()
        addJob()
        addProcess()
        updateProcess()
        addTask()
        getUserConfigurationData()
        getProcess()
        getProcessList()
        getProcessStatus()
        getJob()
        getProcessOutputs()
        getProcessIds()
        getApplicationDeployments()

- Modified _ProcessContext_ and _OrchestratorContext_ to use RegistryServiceClient instead of _AppCatalog_ and _ExperimentCatalog_ objects.

### CODE REVIEW

File: modules/orchestrator/orchestrator-service/src/main/java/org/apache/airavata/orchestrator/server/OrchestratorServerHandler.java [hunk 1]
```diff
@@ -187,16 +182,18 @@ public boolean launchExperiment(String experimentId, String gatewayId) throws TE
 								} else {
 									log.error("Could not find a replica for the URI " + pi.getValue());
 								}
-							} catch (ReplicaCatalogException e) {
-								log.error(e.getMessage(), e);
-							}
-						} else if (pi.getType().equals(DataType.URI_COLLECTION) && pi.getValue().contains("airavata-dp://")) {
+							} catch (RegistryServiceException e) {
+                                e.printStackTrace();
```
Comment (machristie) [inline modules/orchestrator/orchestrator-service/src/main/java/org/apache/airavata/orchestrator/server/OrchestratorServerHandler.java]: Please wrap these in RuntimeException

File: modules/orchestrator/orchestrator-service/src/main/java/org/apache/airavata/orchestrator/server/OrchestratorServerHandler.java [hunk 2]
```diff
@@ -211,14 +208,16 @@ public boolean launchExperiment(String experimentId, String gatewayId) throws TE
 									}
 								}
 								pi.setValue(StringUtils.join(filePathList, ','));
-							} catch (ReplicaCatalogException e) {
-								log.error(e.getMessage(), e);
-							}
-						}
+							} catch (RegistryServiceException e) {
+                                e.printStackTrace();
```
Comment (machristie) [inline modules/orchestrator/orchestrator-service/src/main/java/org/apache/airavata/orchestrator/server/OrchestratorServerHandler.java]: These too

File: modules/orchestrator/orchestrator-service/src/main/java/org/apache/airavata/orchestrator/server/OrchestratorServerHandler.java [hunk 3]
```diff
@@ -538,9 +521,8 @@ public void onMessage(MessageContext message) {
 									status.setState(ExperimentState.EXECUTING);
 									status.setReason("process  started");
 								}
-							} catch (RegistryException e) {
-								status.setState(ExperimentState.EXECUTING);
-								status.setReason("process  started");
+							} catch (ApplicationSettingsException e) {
+								e.printStackTrace();
```
Comment (machristie) [inline modules/orchestrator/orchestrator-service/src/main/java/org/apache/airavata/orchestrator/server/OrchestratorServerHandler.java]: These too.

File: modules/registry/registry-server/registry-api-stubs/src/main/java/org/apache/airavata/registry/api/exception/RegistryServiceException.java [hunk 4]
```diff
@@ -15,15 +15,15 @@
...
+ * Autogenerated by Thrift Compiler (0.11.0)
  *
  * DO NOT EDIT UNLESS YOU ARE SURE THAT YOU KNOW WHAT YOU ARE DOING
  *  @generated
  */
 package org.apache.airavata.registry.api.exception;
 
 @SuppressWarnings({"cast", "rawtypes", "serial", "unchecked", "unused"})
-@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.10.0)")
+@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.11.0)")
```
Comment (machristie) [inline modules/registry/registry-server/registry-api-stubs/src/main/java/org/apache/airavata/registry/api/exception/RegistryServiceException.java]: We're still on version 0.10.0, right?

**Comment by sachinkariyattin** [comment]
@machristie Thanks for the review. Implemented the suggested changes.

**Pull request merged**